src/Symfony/Component/Validator/Attribute/ValidationFor.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Attribute;
+
+/**
+ * Declares that constraints listed on the current class should be added to the given class.
+ *
+ * Classes that use this attribute should contain properties and methods that
+ * exist on the target class.
```

Comment:
I feel like this encourages people to declare _all_ properties and methods of the target class in the source, when the compiler pass only checks that source properties and source methods are available in the target. It's a slight difference, but maybe it should be reworded to explain that only overriding props/methods should be declared?